### PR TITLE
refactor: Address more type issues related to credentials and localization

### PR DESCRIPTION
**Pull request by mutdmour**
## Summary
Address more type issues related to credentials and localization

## Related tickets and issues
TS cleanup

## Review / Merge checklist
- [X] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: packages/editor-ui/src/plugins/i18n/index.ts [hunk 1]
```diff
@@ -548,6 +548,18 @@ type GetBaseTextKey<T> = T extends `_${string}` ? never : T;
...
+function isBaseTextKey(key: string): key is BaseTextKey {
+	return englishBaseText.hasOwnProperty(key);
+}
+
+export function getBaseTextKey(key: string): BaseTextKey {
+	if (isBaseTextKey(key)) {
+		return key;
+	}
+
+	return 'generic.unknown_translation_key';
```
Comment (mutdmour) [inline packages/editor-ui/src/plugins/i18n/index.ts:560]: create fallback key if translation key is unknown.

File: packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue [hunk 2]
```diff
@@ -412,11 +414,10 @@ export default defineComponent({
 				}
 
 				if (property.type === 'number') {
-					const isExpression =
-						typeof this.credentialData[property.name] === 'string' &&
-						this.credentialData[property.name].startsWith('=');
+					const paramValue = this.credentialData[property.name];
+					const isPramExpression = typeof paramValue === 'string' && paramValue.startsWith('=');
```
Comment (elsmr) [inline packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue:418]: typo `isParamExpression`, there is also a util for this: `isExpression()`

File: packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue [hunk 3]
```diff
@@ -414,8 +414,8 @@ export default defineComponent({
 				}
 
 				if (property.type === 'number') {
-					const paramValue = this.credentialData[property.name];
-					const isPramExpression = typeof paramValue === 'string' && paramValue.startsWith('=');
+					const value = this.credentialData[property.name];
+					const isPramExpression = isExpression(value);
```
Comment (elsmr) [inline packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue:418]: Still a typo `Pram` -> `Param`

Comment (mutdmour) [reply packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue]: thanks. fixed. 🙏🏽

**Review by elsmr — DISMISSED** [review]

**Review by elsmr — APPROVED** [review]

**Pull request closed without merging**